Datamodel and mappers

Mefjuu94 opened the pull request:
*Mappers
- MapAbilityBonus -> done, mapping is done by "MapRace" class, passing ArrayNode argument.
*Tests -> Added Folder for better Maintain Tests->  added tests to all mappers classes ->
- MapAbilityBonusTests
- MapCharacterTests
- MapFeaturesTests
- MapRaceTests
- MapSpellsTests

- AbilityBonus -> Added Equals method

Mefjuu94 committed:
*Added sketch of classes:
- AbilityBonus
- Feature
- Item
- Race
- Spells
- Character
*Added JSONMapper who converts Json String to Java Object for create Character class from scratch.

Mefjuu94 committed:
update to main banch

Mefjuu94 committed:
dataModelAndMappers commit

Mefjuu94 committed:
*Created new package with classes to map from JSON to Java Objects: ("+" ready/ "-" in progress)
- MapAbilityBonus -
- MapItems -
- MapCharacterClass +
- MapFeatures +
- MapRace +
- MapSpells +
* Make "JSONMapper" class with one static variable to better maintain code.
* Fixed typos.
* Deleted unnecessary files.

Mefjuu94 committed:
*update
* Deleted unnecessary file "MapItems" -shouldn't be created.

Mefjuu94 committed:
*Created Tests For Mappers:
- MapFeaturesTests + (done main happy and unhappy path)
- MapCharacterTests + (done main happy and unhappy path)
- MapRaceTests + (done main happy and unhappy path)
- MapSpellsTests + (done main happy and unhappy path)
- MapAbilityBonus - (in progress)

Mefjuu94 committed:
*Mappers
- MapAbilityBonus -> done, mapping is done by "MapRace" class, passing ArrayNode argument.
*Tests -> Added Folder for better Maintain Tests->  added tests to all mappers classes ->
- MapAbilityBonusTests
- MapCharacterTests
- MapFeaturesTests
- MapRaceTests
- MapSpellsTests

- AbilityBonus -> Added Equals method

blazejprzyluski commented on build.gradle:
```diff
@@ -9,15 +9,23 @@ repositories {
 
 test {
     useJUnitPlatform()
+
```
unnecessary empty space

Mefjuu94 replied:
corrected!

blazejprzyluski commented on src/main/java/org/example/Character/Mappers/MapCharacterClass.java:
```diff
@@ -0,0 +1,49 @@
+        try {
+            JsonNode jsonNode = MAPPER.readTree(JSON);
+
+            List<String> avaibleSkills = new ArrayList<>();
```
availableSkills

Mefjuu94 replied:
My bad, corrected!

blazejprzyluski commented on src/main/java/org/example/Character/Mappers/MapFeatures.java:
```diff
@@ -0,0 +1,24 @@
+    public ArrayList<Feature> mapClassFeatures(String JSON){
+        try {
+            JsonNode jsonNode = MAPPER.readTree(JSON);
+            ArrayList<Feature> featureList = new ArrayList<>();
```
List on the left side please

Mefjuu94 replied:
Don't get it what's wrong here?

blazejprzyluski replied:
Use List<Feature> featureList = new ArrayList<>();

Mefjuu94 replied:
Corrected!

blazejprzyluski commented on src/main/java/org/example/Character/Mappers/MapSpells.java:
```diff
@@ -0,0 +1,25 @@
+
+public class MapSpells {
+
+    //TODO OGARNĄĆ KTÓrE TO SPELE KLASOWE
```
is this TODO item done?

Mefjuu94 replied:
Yes, corrected!

blazejprzyluski commented on src/main/java/org/example/Character/Spell.java:
```diff
@@ -0,0 +1,25 @@
+
+public class Spell {
+
+    String spellName;
```
private

Mefjuu94 replied:
corrected!

blazejprzyluski commented on src/test/java/ApiHandlersConnectionMockitoTests/ApiHandlersConnectionMockitoTest.java:
```diff
@@ -17,30 +17,30 @@
 
     @Mock
-    private HttpResponse<String> response;
+    public  HttpResponse<String> response;
```
why public?

Mefjuu94 replied:
Because I forgot when I was trying to make it in separate package, but it didnt worked. Of course corrected!

blazejprzyluski commented on src/test/java/ApiHandlersConnectionMockitoTests/ApiHandlersConnectionMockitoTest.java:
```diff
@@ -17,30 +17,30 @@
+    public  HttpResponse<String> response;
     @Mock
-    private HttpClient client;
+    public  HttpClient client;
```
why public?

Mefjuu94 replied:
Because I forgot when I was trying to make it in separate package, but it didnt worked. Of course corrected!

blazejprzyluski commented on src/test/java/ApiHandlersConnectionMockitoTests/ApiHandlersConnectionMockitoTest.java:
```diff
@@ -17,30 +17,30 @@
+    public  HttpClient client;
     @Mock
-    private HttpRequest request;
+    public  HttpRequest request;
```
why public?

Mefjuu94 replied:
Because I forgot when I was trying to make it in separate package, but it didnt worked. Of course corrected!

blazejprzyluski commented on src/test/java/MapersTests/MapAbilityBonusTests.java:
```diff
@@ -0,0 +1,39 @@
+    private final MapAbilityBonus testObject = new MapAbilityBonus();
+    private final ArrayNode badOutputMapper = (ArrayNode) JSONMapper.MAPPER.readTree("[{\"ability_score\":{\"index\":\"dex\",\"name\":\"DEX\",\"url\":\"/api/ability-scores/dex\"},\"noslda\":2}]");
+
+    public MapAbilityBonusTests() throws JsonProcessingException {
```
Why constructor?

Mefjuu94 replied:
Because it throws "JsonMappingException", I can solve this by
- add exception to class default counstructor signature
- surround with try/catch

blazejprzyluski replied:
Try catch is better. You won't ever use this constructor

Mefjuu94 replied:
ok, corrected!

blazejprzyluski commented on src/test/java/MapersTests/MapAbilityBonusTests.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    @Test
+    public void mapperUnHappyPathNullPointerException(){
```
NullPointer is not thrown in this test method

Mefjuu94 replied:
Yes, corrected!

blazejprzyluski commented on src/test/java/MapersTests/MapCharacterTests.java:
```diff
@@ -0,0 +1,36 @@
+    @Test
+    public void mapperUNHappyPathNull() {
+        String resultForString = testObject.mapCharacterClass(outputResponseBody).toString();
+        Assertions.assertNotEquals(null, resultForString);
```
Checking for null is not a good check

Mefjuu94 replied:
Better is empty string?

blazejprzyluski replied:
Anything could be null, or an empty String. This does not test anything specific, same as other places

Mefjuu94 replied:
ok, corrected!

blazejprzyluski commented on src/test/java/MapersTests/MapCharacterTests.java:
```diff
@@ -0,0 +1,36 @@
+
+    @Test
+    public void mapperUNHappyPath() {
+        Assertions.assertNotEquals(outputMapper, " ");
```
what are you checking here?

Mefjuu94 replied:
I forgot to delete an unnecessary test to check if the tests are working correctly. corrected!

blazejprzyluski commented on src/test/java/MapersTests/MapFeaturesTests.java:
```diff
@@ -0,0 +1,35 @@
+
+    @Test
+    public void mapperUNHappyPath() {
+        Assertions.assertNotEquals(outputMapper, " ");
```
Same as in character test

Mefjuu94 replied:
Corrected!

blazejprzyluski commented on src/test/java/MapersTests/MapRaceTests.java:
```diff
@@ -0,0 +1,35 @@
+
+    @Test
+    public void mapperUNHappyPath() {
+        Assertions.assertNotEquals(outputMapper, " ");
```
same thing for testing. This is not a strong check

Mefjuu94 replied:
Corrected!

blazejprzyluski commented on src/test/java/MapersTests/MapSpellsTests.java:
```diff
@@ -0,0 +1,33 @@
+
+    @Test
+    public void mapperUNHappyPath() {
+        Assertions.assertNotEquals(outputMapper, " ");
```
as in the rest of tests

Mefjuu94 replied:
Corrected!

Mefjuu94 committed:
Fixed few issues after pull request:
- deleted some empty unnecessary spaces.
- deleted wrong tests.
- change Field on private in Spell class.

Mefjuu94 committed:
Fixed few issues after pull request:
- deleted unnecessary Tests in mapTests classes.
- making Feature now return List instead of ArrayList
- surrounding by try/catch block instead of adding exception to class default counstructor signature in MapAbilityBonusTests
